### ISSUE TITLE
apigee_environment_kvm -> using encrypted values showing the secret

### ISSUE THREAD

**Issue by aglamela** [scastria/terraform-provider-apigee#11]
Hi, great job with this provider!

We are starting to use it, and we have found that when using the resource apigee_environment_kvm, and when we enable encrypted variable to true, it is showing the value although it is a "secret" or a variable that is considered as a secret in apigee... Would it be any way to avoid showing the variable or masking the value when enabling the encrypted variable?

**Comment by scastria** [comment]
Where do you see the secret value?  In the state file, in the log, etc?

**Comment by aglamela** [comment]
In the log when doing terraform plan.

**Comment by scastria** [comment]
I do have Sensitive = true for the values so not sure why TF is not honoring that.  Let me check some more...

**Comment by scastria** [comment]
I had set Sensitive = true ONLY for the value of the KVM entries, not the keys.  I guess TF providers do not allow that.  If I move the Sensitive = true up for the entire entry property, it works.  However, TF providers cannot be dynamic about whether a property is sensitive or not.  So there is no way for me to use the value of encrypted property to determine the sensitivity of the entry property.

Do you have any suggestions?

I could add a second property called sensitive_entry to go along with entry.  I could prevent both entry and sensitive_entry from being specified.  The sensitive_entry is hardcoded to be sensitive and entry is hardcoded NOT to be sensitive.  Then you could use sensitive_entry when you needed privacy in the logs?  This solution doesn't sound too bad to me.  What do you think?

**Comment by aglamela** [comment]
Hi,

What I mean, we use a kms to put our password encrypted in our code repos. So, what I don't like to see later those encrypted password in clear in the logs

```
data "google_kms_secret" "myaccount_Password" {
  crypto_key = "europe-west1/kms-apigee/key-apigee"
  ciphertext = "CiQA2Ur9dbkENz9c3cU152S+6Beai2YvX4h+ZPB4afhPuZOv3R0SOgDG8M25ufMXUdpU9gAxGE+5ukgWdFr6Nk4KHyzoSdG5XRBSg9JRB5HmYJ5UOhrlsAjlmA6GGiXenyU="
}

resource "apigee_environment_kvm" "kvm_secrets_myaccount" {
  environment_name = "test"
  name = "myaccount_secrets"
  entry = {
    myaccount_Password = data.google_kms_secret.myaccount_Password.plaintext
  }
  encrypted = true
}

```

```
#LOG
 # apigee_environment_kvm.kvm_secrets_myaccount will be created
  + resource "apigee_environment_kvm" "kvm_secrets_myaccount" {
      + encrypted        = true
      + entry            = {
          + "myaccount_Password" = *"my-secret-user"*
        }
      + environment_name = "test"
      + id               = (known after apply)
      + name             = "myaccount_secrets"
    }
```

What you propose I think is nice, but what you think instead of having to specify sensitive_entry or entry, using entry, and use the encrypted key to implement this logic?

**Comment by scastria** [comment]
I cannot do that.  As I stated above, "However, TF providers cannot be dynamic about whether a property is sensitive or not. So there is no way for me to use the value of encrypted property to determine the sensitivity of the entry property."

The only 2 options I see are:
1.  Make entry ALWAYS hide the KVM values in the log
2. Introduce a new separate sensitive_entry that ALWAYS hides the values in the log and keep entry that NEVER hides the values in the log.

Since I don't like option 1, I guess I will do option 2.

**Comment by aglamela** [comment]
Thank you very much for your support!!!

**Comment by scastria** [comment]
v0.1.35 is building now.  Keep in mind that the Apigee API does NOT allow reading back encrypted KVM values.  Therefore, TF cannot properly compare your TF config to the actual KVM inside Apigee and will always think there is a change to your TF config that needs to be applied.  There is nothing I can do about that.  So your only option is to use the:
```
lifecycle {
  ignore_changes = [sensitive_entry]
}
```
to suppress these false changes.  But then, of course, when you DO make real changes, you have to remove the lifecycle section so that you can apply those changes and then put the lifecycle section back in.

**Comment by scastria** [comment]
I just did a migration test from v0.1.34 to v0.1.35 and it DID work smoothly.  Here was my test:
1. initialized with v0.1.34
2. Created a KVM using `encrypted` = `true` and using `entry` to specify values
3. applied
4. upgrade to v0.1.35
5. applied.  Since I didn't change `entry` to `sensitive_entry`, it deleted all KVM values from Apigee
6. I then changed `entry` to `sensitive_entry`
7. applied 
8. Everything was corrected and I had KVM values hidden in the log.

This is a successful test.  I would be nice if I add a WARNING message stating that `sensitive_entry` should be used when `encrypted` = `true`
